iree/turbine/kernel/ops/wave_ops.py

```diff
@@ -721,6 +729,10 @@ def type(self) -> Any:
     def type(self, value: Any):
         self.fx_node.type = value
 
+    @property
+    def skip_dce(self) -> bool:
```

Comment:
Maybe something like `has_side_effects`? Also I think it will be safer to set `has_side_effects` to True by default and then set to False for the ops we know have no effects.

Reply:
that will be all the ops then hahaha

Reply:
I agree with the rename, but I think we should keep default to false. The reasoning behind it being most ops we work with has no side effects and people adding ops are usually compiler author who should/would know to add this attribute when necessary. :)